### PR TITLE
ENG-1088 Marketplace bug fixes

### PR DESCRIPTION
**Pull request by Motokoder**
- Added marketplace context provider to lift state so the main page can load quickly when navigating back.
- Updated Filter/Sort component to load initial filter/sort values passed as args.
  - Values from context are passed in to restore drop-down selections when navigating back.
- Added TypeScript interfaces.
- Fixed filtering and sorting.
- Refactored code to flatten nested structures and use async/await in place of promise callbacks.
- Renamed "NFT" to "Digital Certificate" or "ODC" (Origyn Digital Certificate) in all forms.
- Removed dead code.

### CODE REVIEW

**Review by frank0170 — APPROVED** [review]

File: packages/apps/marketplace/src/pages/Marketplace/index.tsx [hunk 1]
```diff
@@ -266,47 +268,62 @@ const Marketplace = () => {
...
+                          );
                         })}
                       </Grid>
                       <br />
                     </>
                   ) : (
-                    <h5>
-                      There are no NFTs in this collection
-                    </h5>
+                    <h5>There are no NFTs in this collection</h5>
```
Comment (skilesare) [inline packages/apps/marketplace/src/pages/Marketplace/index.tsx:326]: Change NFTs to Digital Certificates.

**Review by skilesare — CHANGES_REQUESTED** [review]
Lets write clean code.  There are a ton of GitHub Actions
/ Code Quaility & Tests warnings and errors.  Please clean as many of these up and figure out the ignore command for ones you can't clear and then resubmit the pull request.

**Comment by Motokoder** [comment]
> Lets write clean code. There are a ton of GitHub Actions / Code Quaility & Tests warnings and errors. Please clean as many of these up and figure out the ignore command for ones you can't clear and then resubmit the pull request.

@skilesare Yes, we are on it. Those are all existing linting errors that I will be working on today as a separate PR. No new linting errors are introduced in the PR. Linting was getting skipped because a previous GitHub action to install dependencies was failing due to auth issues when installing the UI Component library (because it's not published on npm yet). The only reason for creating the PR is to get a review and test before deploying the latest (beta) marketplace to the Yumi canisters with performance improvements and bug fixes.

**Review by skilesare — APPROVED** [review]

**Review by oleg-kkk — APPROVED** [review]

File: packages/apps/marketplace/src/pages/Marketplace/index.tsx (hunk 1 above)
Comment (Motokoder) [reply packages/apps/marketplace/src/pages/Marketplace/index.tsx]: Done.

**Pull request merged**